postreise/plot/plot_scatter_capacity_vs_capacity_factor.py

```diff
@@ -49,21 +50,33 @@ def plot_scatter_capacity_vs_capacity_factor(
     :param int/float fontsize: font size, default to 20.
     :param str title: user specified figure title, default to None.
     :param bool percentage: show capacity factor in percentage or not, default to False
-    :param bool show_plot: show the plot or not, default to True.
+    :param bool plot_show: show the plot or not, default to True.
     :return: (*tuple*) -- the first entry is matplotlib.axes.Axes object of the plot,
         the second entry is the capacity weighted average of capacity factors over the
         selected time range.
     :raises TypeError:
-        if markersize is not an integer or a float and/or
-        if fontsize is not an integer or a float and/or
-        if title is provided but not in a string format.
+        if ``area`` is not a str.
+        if ``resources`` is not a str or a list of str.
+        if ``time_zone`` is not a str.
+        if ``markersize`` is not a int or a float.
+        if ``fontsize`` is not a int or a float.
+        if ``title`` is provided but not a str.
     """
+    _check_scenario_is_in_analyze_state(scenario)
+
+    if not isinstance(area, str):
+        raise TypeError("area must be a str")
+    if not isinstance(resources, (str, list)):
+        raise TypeError("resources must be a list or str")
+    if isinstance(resources, list) and not all([isinstance(r, str) for r in resources]):
```

Comment:
I think the `[]` are not needed: `all(isinstance(r, str) for r in resources)`

Reply:
Indeed

Reply:
Done